app/app.go

```diff
@@ -145,19 +154,26 @@ var (
 		vesting.AppModuleBasic{},
 		liquidity.AppModuleBasic{},
 		router.AppModuleBasic{},
+		ica.AppModuleBasic{},
 	)
 
 	// module account permissions
 	maccPerms = map[string][]string{
 		authtypes.FeeCollectorName:     nil,
 		distrtypes.ModuleName:          nil,
+		icatypes.ModuleName:            nil,
 		minttypes.ModuleName:           {authtypes.Minter},
 		stakingtypes.BondedPoolName:    {authtypes.Burner, authtypes.Staking},
 		stakingtypes.NotBondedPoolName: {authtypes.Burner, authtypes.Staking},
 		govtypes.ModuleName:            {authtypes.Burner},
 		liquiditytypes.ModuleName:      {authtypes.Minter, authtypes.Burner},
 		ibctransfertypes.ModuleName:    {authtypes.Minter, authtypes.Burner},
 	}
+
+	// module accounts that are allowed to receive tokens
+	allowedReceivingModAcc = map[string]bool{
```

Comment:
before, we block all module account addresses to receive tokens by `MsgSend`, but [interchain account](https://github.com/cosmos/interchain-accounts/blob/7a4ade49c4ddc74b44e23c44906be371d00a322b/app/app.go#L547) do not block distribution module (which scaffold from starport), should we keep blocking all the address or make distribution an exception ?

Reply:
we actually allow all modules in gaia https://github.com/cosmos/gaia/blob/v6.0.3/app/app.go#L653-L661

Reply:
Won't allowing the distribution module to receive tokens break [invariants](https://github.com/cosmos/cosmos-sdk/blob/26d573e3dab889bb5527a454b1a6437a8760c263/x/distribution/keeper/invariants.go#L138)? While not well documented, this was the intention of the blocked addresses, to prevent invariants from being broken